libregexp.c

```diff
@@ -1961,86 +1961,86 @@ static BOOL is_word_char(uint32_t c)
             (c == '_'));
 }
 
-#define GET_CHAR(c, cptr, cbuf_end)                                     \
+#define GET_CHAR(c, cptr, cbuf_end, cbuf_type)                          \
     do {                                                                \
         if (cbuf_type == 0) {                                           \
             c = *cptr++;                                                \
         } else {                                                        \
-            const uint16_t *_p = (uint16_t *)cptr;                      \
-            const uint16_t *_end = (uint16_t *)cbuf_end;                \
+            const uint16_t *_p = (const uint16_t *)cptr;                \
+            const uint16_t *_end = (const uint16_t *)cbuf_end;          \
             c = *_p++;                                                  \
-            if (is_hi_surrogate(c))                                     \
-                if (cbuf_type == 2)                                     \
-                    if (_p < _end)                                      \
-                        if (is_lo_surrogate(*_p))                       \
-                            c = from_surrogate(c, *_p++);               \
-            cptr = (void *) _p;                                         \
+            if (is_hi_surrogate(c) && cbuf_type == 2) {                 \
+                if (_p < _end && is_lo_surrogate(*_p)) {                \
+                    c = from_surrogate(c, *_p++);                       \
+                }                                                       \
+            }                                                           \
```

Comment:
Can you leave out stylistic changes? They make the diff larger and more difficult to read.

As a reviewer, I first have to figure out if it's actually a functional change - and if not, if it's really equivalent to the old code. As a maintainer, style changes introduce churn that makes `git blame` less effective. As a contributor, it's race inducing when your carefully crafted pull request ends up with a ton of merge conflicts because someone landed a style change the other day.

Changes that improve legibility are okay but a) sparingly, and b) should be split off into separate commits and marked NFC (Non-Functional Change.)

Reply:
Sorry about that, I was trying to reduce the macro size and combine the tests for consistency. I posted a patch with minimal NFC differences.